tdms/src/interpolation_methods.cpp

```diff
@@ -30,29 +30,29 @@ double interp3(double *v)
     return 1. / 16. * v[0] - 5. / 16. * v[1] + 15. / 16. * v[2] + 5. / 16. * v[3];
 }
 
-void checkInterpolationPoints(int i_l, int i_u, int j_l, int j_u, int k_l, int k_u, int I, int J, int K)
+void checkInterpolationPoints(int i_l, int i_u, int j_l, int j_u, int k_l, int k_u, int nI, int nJ, int nK)
 {
     if (i_l < 2)
     {
         throw runtime_error("Interpolation error: i_l too small");
     }
-    else if (i_u > I - 2)
+    else if (i_u > nI - 2)
     {
         throw runtime_error("Interpolation error: i_u too large");
     }
     else if (j_l < 2)
     {
         throw runtime_error("Interpolation error: j_l too small");
     }
-    else if (j_u > J - 2)
+    else if (j_u > nJ - 2)
     {
         throw runtime_error("Interpolation error: j_u too large");
     }
     else if (k_l < 2)
     {
         throw runtime_error("Interpolation error: k_l too small");
     }
-    else if (k_u > K - 2)
+    else if (k_u > nK - 2)
     {
         throw runtime_error("Interpolation error: k_u too large");
     }
```

Comment:
Below, int the case of CUBIC_INTERP_FIRST, and the other cubic schemes, why is:

last_nonzero_coeff = 4;

when it looks (to my current understanding) as though it should be:

last_nonzero_coeff = 3;

Reply:
They should be 3, this is me being guilty of copy/paste without checking everything